### ISSUE TITLE
golangci-lint in superlinter do not scan properly many files in package

### ISSUE THREAD

**Issue by asiekkowa** [github/super-linter#1599]
## Describe the bug
golangci-lint do not scan properly many files in package

## Expected behavior
`golangci-lint run -c ../.github/linters/.golangci.yml` doesn't show any error. Expected behavior is that super-linter does not show any errors as well.

## Steps to Reproduce

env.go
```
package tests

import (
	"fmt"
	"os"
)

func getEnv(key, fallback string) string {
	if value, ok := os.LookupEnv(key); ok {
		return value
	}
	return fallback
}

var (
	tfVars = map[string]interface{}{
		"suffix":        getEnv("TT_SUFFIX", ""),
		"aws_region":    getEnv("TT_REGION", "us-east-1"),
		"environment":   getEnv("TT_ENV", "pr"),
	}

	backendConfig = map[string]interface{}{
		"role_arn":       os.Getenv("TT_BACKEND_ROLE"),
		"encrypt":        os.Getenv("TT_BACKEND_ENCRYPT"),
		"bucket":         os.Getenv("TT_BACKEND_BUCKET"),
		"key":            os.Getenv("TT_BACKEND_KEY"),
		"dynamodb_table": os.Getenv("TT_BACKEND_DYNAMODB"),
		"region":         os.Getenv("TT_BACKEND_REGION"),
	}


	tfDir = "../"

	varFiles = getEnv("TT_VAR_FILE_PATH",
		fmt.Sprintf("envs/%s/%s.tfvars", tfVars["environment"], tfVars["aws_region"]))
)
```

test_plan.go
```
package tests

import (
	"github.com/gruntwork-io/terratest/modules/terraform"
	"testing"
)

func TestTerraformTestOutput(t *testing.T) {
	t.Parallel()

	terraformOptions := &terraform.Options{
		TerraformDir:  tfDir,
		BackendConfig: backendConfig,
		VarFiles:      []string{varFiles},
		Vars:          tfVars,
		// Disable colors in Terraform commands so its easier to parse stdout/stderr
		NoColor: true,
	}

	terraform.InitAndPlan(t, terraformOptions)
}
```

```
docker run -e RUN_LOCAL=true -v $(pwd):/tmp/lint -e VALIDATE_GO=true -e ACTIONS_RUNNER_DEBUG=true  ghcr.io/github/super-linter:latest
```

## Logs

```
2021-05-27 13:31:38 [DEBUG]   Linting FILE: /tmp/lint/tests/env.go
2021-05-27 13:31:38 [DEBUG]   FILE_STATUS (env.go) is empty, or not set to 'good' or 'bad'. Assuming it should be linted correctly. Setting FILE_STATUS to 'good'...
2021-05-27 13:31:38 [DEBUG]   File: /tmp/lint/tests/env.go, FILE_NAME: env.go, DIR_NAME:/tmp/lint/tests, FILE_STATUS: good, INDIVIDUAL_TEST_FOLDER: go
2021-05-27 13:31:38 [INFO]   ---------------------------
2021-05-27 13:31:38 [INFO]   File:[/tmp/lint/tests/env.go]
2021-05-27 13:31:38 [DEBUG]   Found errors. Error code: 1, File type: GO, Error on missing exec bit: false
2021-05-27 13:31:38 [ERROR]   Found errors in [golangci-lint] linter!
2021-05-27 13:31:38 [ERROR]   Error code: 1. Command output:
------
tests/env.go:25:2: `backendConfig` is unused (deadcode)
	backendConfig = map[string]interface{}{
	^
tests/env.go:37:2: `tfDir` is unused (deadcode)
	tfDir = "../"
	^
tests/env.go:39:2: `varFiles` is unused (deadcode)
	varFiles = getEnv("TT_VAR_FILE_PATH",
	^
------
2021-05-27 13:31:38 [DEBUG]   Error code: 1. Command output:
------
tests/env.go:25:2: `backendConfig` is unused (deadcode)
	backendConfig = map[string]interface{}{
	^
tests/env.go:37:2: `tfDir` is unused (deadcode)
	tfDir = "../"
	^
tests/env.go:39:2: `varFiles` is unused (deadcode)
	varFiles = getEnv("TT_VAR_FILE_PATH",
	^
------
2021-05-27 13:31:39 [DEBUG]   Linting FILE: /tmp/lint/tests/tfplan_test.go
2021-05-27 13:31:39 [DEBUG]   FILE_STATUS (test.go) is empty, or not set to 'good' or 'bad'. Assuming it should be linted correctly. Setting FILE_STATUS to 'good'...
2021-05-27 13:31:39 [DEBUG]   File: /tmp/lint/tests/tfplan_test.go, FILE_NAME: tfplan_test.go, DIR_NAME:/tmp/lint/tests, FILE_STATUS: good, INDIVIDUAL_TEST_FOLDER: go
2021-05-27 13:31:39 [INFO]   ---------------------------
2021-05-27 13:31:39 [INFO]   File:[/tmp/lint/tests/tfplan_test.go]
2021-05-27 13:31:39 [DEBUG]   Found errors. Error code: 1, File type: GO, Error on missing exec bit: false
2021-05-27 13:31:39 [ERROR]   Found errors in [golangci-lint] linter!
2021-05-27 13:31:39 [ERROR]   Error code: 1. Command output:
------
tests/tfplan_test.go:12:18: undeclared name: `tfDir` (typecheck)
		TerraformDir:  tfDir,
		               ^
tests/tfplan_test.go:13:18: undeclared name: `backendConfig` (typecheck)
		BackendConfig: backendConfig,
		               ^
tests/tfplan_test.go:14:27: undeclared name: `varFiles` (typecheck)
		VarFiles:      []string{varFiles},
		                        ^
tests/tfplan_test.go:15:18: undeclared name: `tfVars` (typecheck)
		Vars:          tfVars,
		               ^
------
2021-05-27 13:31:39 [DEBUG]   Error code: 1. Command output:
------
tests/tfplan_test.go:12:18: undeclared name: `tfDir` (typecheck)
		TerraformDir:  tfDir,
		               ^
tests/tfplan_test.go:13:18: undeclared name: `backendConfig` (typecheck)
		BackendConfig: backendConfig,
		               ^
tests/tfplan_test.go:14:27: undeclared name: `varFiles` (typecheck)
		VarFiles:      []string{varFiles},
		                        ^
tests/tfplan_test.go:15:18: undeclared name: `tfVars` (typecheck)
		Vars:          tfVars,
		               ^
------


```

**Comment by admiralAwkbar** [comment]
@asiekkowa Will look into this... Can you tell me what version of `golangci-lint` you have installed locally?

**Comment by admiralAwkbar** [comment]
@asiekkowa Im also getting this error when trying to run the linter....

```bash
bash-5.1# cd test/
bash-5.1# golangci-lint run
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package : could not load export data: no export data for "github.com/gruntwork-io/terratest/modules/terraform"
bash-5.1#
```

**Comment by asiekkowa** [comment]
> @asiekkowa Will look into this... Can you tell me what version of `golangci-lint` you have installed locally?

golangci-lint has version 1.40.1 built from 625445b on 2021-05-14T16:27:59Z

**Comment by angelbarrera92** [comment]
I hit the same issue. Seems like the linter runs on single files so the package is not computed...
Is there any workaround?

**Comment by Hades32** [comment]
@admiralAwkbar was this fixed? Just as @angelbarrera92 I get errors (without any golang-ci config, so just using the defaults) about undefined types etc that ARE defined, but just in other files within the same package...

**Comment by indrora** [comment]
I am similarly seeing this in a large (600+ file) repo I have that we're trying to add linting to. A certain amount of it is code that was written before the days of modules and is kept for historical reasons. 

It appears that something to do with how golangci-lint handles modules is funny when run under the superlinter: we have probably a dozen or so nested modules within each other, and I suspect something is getting out of alignment between them.

Similarly, because of how Golangci-lint is being called, it lints files I don't want linted.
